cmds/tail/tail_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+)
+
+func TestTailReadBackwards(t *testing.T) {
+	input, err := os.Open("./test_samples/read_backwards.txt")
+	if err != nil {
+		t.Error(err)
+	}
+	output := bytes.NewBuffer([]byte{})
+	err = readLastLinesBackwards(*input, output, 2)
+	if err != nil {
+		t.Error(err)
+	}
+	expected := []byte("second\nthird\n")
+	got := make([]byte, 4096) // anything larger than the expected result
+	n, err := output.Read(got)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(expected, got[:n]) {
+		t.Fatalf("Invalid result reading backwards. Expected %v, got %v", expected, got[:n])
+	}
+	// try reading more, which should return EOF
+	buf := make([]byte, 16)
+	n, err = input.Read(buf)
+	if err == nil {
+		t.Fatalf("Expected EOF, got more bytes: %v", string(buf[:n]))
+	}
+	if err != io.EOF {
+		t.Fatalf("Expected EOF, got another error instead: %v", err)
+	}
+}
+
+func TestTailReadFromBeginning(t *testing.T) {
+	input, err := os.Open("./test_samples/read_from_beginning.txt")
+	if err != nil {
+		t.Error(err)
+	}
+	output := bytes.NewBuffer([]byte{})
+	err = readLastLinesBackwards(*input, output, 3)
+	if err != nil {
+		t.Error(err)
+	}
+	expected := []byte("eight\nnine\nten\n")
+	got := make([]byte, 4096) // anything larger than the expected result
+	n, err := output.Read(got)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(expected, got[:n]) {
+		t.Fatalf("Invalid result reading backwards. Expected %v, got %v", string(expected), string(got[:n]))
```

Comment:
Update the messaging here to indicate you are trying to read from the beginning.